NossoMercadoLivre.Domain/Common/Helper.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace NossoMercadoLivre.Domain.Common
+{
+    public static class Helper
+    {
+        public static string GetSha256Hash(SHA256 shaHash, string input)
```

Comment:
Parece que tem, esse Sha256 :). Pq ter uma classe a mais?

Reply:
O método é apenas para conversão de uma string em uma string em sha256, para ser mais fácil de salvar em banco. A classe é separada pois não é um método que descreve bem o que a classe faz, seguindo os conceitos de orientação a objeto.

Reply:
Oi Erick, esse método é estático, ele está fora do contexto de orientação a objetos independente de onde você colocar ele :). A classe chama Helper, que é descontextualizada de qualquer problema que você está tentando resolver :). A maioria das aplicações web, tem mais procedure do que classe com estado e comportamento em si, como está escrito no texto que linkei para vocês. 

Por outro lado, eu só percebi agora que o método que faz o encode da string retorna um array de byte e você quis concentrar esse código aqui. O que não precisa é receber como argumento o objeto que faz o hash, deveria ser variável local. Porém....

*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

Tem a classe PasswordHasher do asp.net core 3.1 => https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.identity.passwordhasher-1?view=aspnetcore-3.1 E ela já aplica o hash e retorna o String.